rllib/algorithms/bc/bc.py

```diff
@@ -144,75 +146,62 @@ def training_step(self) -> ResultDict:
-
-            # Updating the policy.
-            train_results = self.learner_group.update_from_batch(batch=train_batch)
-            # TODO (sven): Use metrics API as soon as we moved to new API stack
-            #  (from currently hybrid stack).
-            # self.metrics.log_dict(
-            #    {
-            #        NUM_AGENT_STEPS_TRAINED_LIFETIME: len(train_batch),
-            #        NUM_ENV_STEPS_TRAINED_LIFETIME: len(train_batch),
-            #    },
-            #    reduce="sum",
-            # )
-            self._counters[NUM_AGENT_STEPS_TRAINED] += len(train_batch)
-            self._counters[NUM_ENV_STEPS_TRAINED] += len(train_batch)
-
+                batch = self.offline_data.sample(
+                    num_samples=self.config.train_batch_size,
+                    num_shards=self.config.num_learners,
+                    return_iterator=True if self.config.num_learners > 1 else False,
+                )
+
+            with self.metrics.log_time((TIMERS, LEARNER_UPDATE_TIMER)):
+                # Updating the policy.
+                # TODO (simon, sven): Check, if we should execute directly s.th. like
+                # update_from_iterator.
+                learner_results = self.learner_group.update_from_batch(batch)
+
+                # Log training results.
+                self.metrics.merge_and_log_n_dicts(learner_results, key=LEARNER_RESULTS)
+                self.metrics.log_value(
+                    NUM_ENV_STEPS_TRAINED_LIFETIME,
+                    self.metrics.peek(
+                        (LEARNER_RESULTS, ALL_MODULES, NUM_ENV_STEPS_TRAINED)
+                    ),
+                    reduce="sum",
+                )
+                self.metrics.log_dict(
+                    {
+                        (LEARNER_RESULTS, mid, NUM_MODULE_STEPS_TRAINED_LIFETIME): (
+                            stats[NUM_MODULE_STEPS_TRAINED]
+                        )
+                        for mid, stats in self.metrics.peek(LEARNER_RESULTS).items()
+                    },
+                    reduce="sum",
+                )
             # Synchronize weights.
             # As the results contain for each policy the loss and in addition the
             # total loss over all policies is returned, this total loss has to be
             # removed.
-            policies_to_update = set(train_results.keys()) - {ALL_MODULES}
+            modules_to_update = set(learner_results[0].keys()) - {ALL_MODULES}
 
-            # with self.metrics.log_time((TIMERS, SYNCH_WORKER_WEIGHTS_TIMER)):
-            with self._timers[SYNCH_WORKER_WEIGHTS_TIMER]:
+            # Update weights and global_vars - after learning on the local worker -
+            # on all remote workers.
+            with self.metrics.log_time((TIMERS, SYNCH_WORKER_WEIGHTS_TIMER)):
                 if self.workers.num_remote_workers() > 0:
+                    # NOTE: the new API stack does not use global vars.
```

Comment:
nit: remove this comment entirely (let's not even mention it anymore, just causes confusion for users)